### ISSUE TITLE
Run a11y tests only for affected components

### ISSUE THREAD

**Issue by lukasholzer** [dynatrace-oss/barista#601]
# Feature Request

## Summary

Currently, if anything affects something in the components we are running all a11y tests. This consumes a lot of ci resources because we have around 300 examples that we are testing.

To shorten our CI load we should only run affected examples.

## Feature Description

To accomplish this we should tag our a11y tests with the component name to run them by specific metatdata.

```
fixture(component.name)
  .meta('component', component.name);
```

Then we can run the a11y tests by the affected component name: The list of all affected libs should be retrieved by `nx affected:libs`.

**Comment by tomheller** [comment]
@lukasholzer I see a little bit of an issue with that. 
Right now, we just iterate over all of the example component folders, but these do not necessarily correlate the the nx dependency graph. I.e. a change in the button should also run the table tests, as it is included there and should also serve it's a11y responsibilities there. 

Would that mean that we would have to write the fixtures for every component ourselves, like:  
```ts
fixture('table')
  .meta('component', 'table')
  .meta('component', 'button')
  .meta('component', 'empty-state');
```

**Comment by lukasholzer** [comment]
To clarify it a little bit. When I change something in the empty state the command `nx affected:libs` will output something like `empty-state, table, tree-table, components`. The picture of the dependency graph will hopefully explain that to you.

![image](https://user-images.githubusercontent.com/11156362/74922613-c5cb6b80-53cf-11ea-94c9-84c1643a87a7.png)


This will mean when we run testcafe with `--fixture-component=empty-state=mobile,component=table,component=tree-table`.  The a11y tests of all affected components will run. But we don"t have to run the radio or checkbox a11y tests.

**Comment by tomheller** [comment]
After an initial trial with defining this, we hit a little bit of a bump here. We were able to extract the affected components from a PR or changeset and pass them via a custom builder to the polite testcafe builder ( which runs a11y and e2e tests). 
Unfortunately the `fixtureMeta` filter outlined above is a match all metadata filter and not a a match any. 
So if the filter passed to the builder has multiple components affected, the fixture would only run if it defines more than the affected components. 
Examples: 
* Button component has a metadata of `button` and `icon` and the affected component would be `button`, the tests would run.
* Button component has a metadata of `button` and `icon` and the affected component would be `button`, `expandable` and `table`, the tests for the button would not run, because it does not contain `expandable` or `table`. 

At this point I have no idea how to mitigate this problem.

**Comment by lukasholzer** [comment]
spawn a worker for each test and try to run in the first worker fixtureMeta button, second worker expandable third worker table.

would be maybe an approach. Not sure if it is smart but yea. Then I would decouple the bootstrapping and serving to the app in the main thread and only start the test without the building of the app so the workers just would run the tests.
